singularity-eos/base/variadic_utils.hpp

```diff
@@ -156,7 +181,7 @@ constexpr auto pack_size(type_list<Ts...>) {
   return sizeof...(Ts);
 }
 
-} // namespace detail
+} // namespace types
```

Comment:
This namespace contains more than types, what is the purpose of the namespace name change?

Reply:
I wanted to expose this machinery to downstream codes so they could use the automated stuff to build up a variant. Not married to the naming, and I'm willing to change it back if you prefer.

Reply:
Renamed to `variadic_utils`